src/components/form/HdRange.vue

```diff
@@ -0,0 +1,329 @@
+    <ul v-if="displayStepBullets" class="range__steps">
+      <li v-for="(steps, i) in stepsAmount" :key="i" />
+    </ul>
+    <div class="range__thumb" ref="thumb">
+      <div class="range__thumb__inner" ref="thumbInner">
+        <div class="range__thumb__bullet" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+    displayStepBullets: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  data() {
+    return {
+      isActive: null,
+      currentValue: this.value,
+    };
+  },
+  computed: {
+    fieldClasses() {
+      return {
+        'field--active': this.isActive,
+        'field--disabled': this.disabled,
+      };
+    },
+    stepsAmount() {
+      return 1 + (this.maxValue - this.minValue) / this.rangeStep;
+    },
+  },
+  watch: {
+    minValue() {
+      this.updateRangeDecoration();
+    },
+    maxValue() {
+      this.updateRangeDecoration();
+    },
+    rangeStep() {
+      this.updateRangeDecoration();
+    },
+    value() {
+      this.currentValue = this.value;
+    },
+    currentValue() {
+      this.updateRangeDecoration();
+    },
+  },
+  mounted() {
+    // the progress bar needs a starting fixed width to be then transformed in updateRangeDecoration() with scaleX property
+    this.$refs.progress.style.width = '1px';
+
+    this.updateRangeDecoration();
+  },
+  methods: {
+    normalizeAndEmit() {
+      if (this.currentValue > this.maxValue) {
+        this.currentValue = this.maxValue;
+      } else if (this.currentValue < this.minValue) {
+        this.currentValue = this.minValue;
+      }
+
+      this.currentValue = this.currentValue - (this.currentValue % this.rangeStep);
+
+      this.$emit('input', this.currentValue);
+    },
+    updateRangeDecoration() {
+      this.normalizeAndEmit();
+
+      // the percentage of the value, between max and min. I.e. : 15 is the 50% between 10 and 20
+      const valuePercentage = ((this.currentValue - this.minValue) / (this.maxValue - this.minValue)) * 100;
+      const valuePercentageInputWidthPixels = (this.$refs.decoration.offsetWidth * valuePercentage) / 100;
+
+      this.$refs.progress.style.transform = `scaleX(${valuePercentageInputWidthPixels})`;
+      this.$refs.thumb.style.transform = `translateX(${valuePercentageInputWidthPixels}px)`;
+    },
+    focusHandler() {
+      this.isActive = true;
+    },
+    blurHandler() {
+      this.isActive = false;
+    },
+  },
+};
+</script>
+
+<style scoped lang="scss">
+.range {
```

Comment:
I suggest to explicitly import `mixins.scss` here, so we slowly shift away from the patter of having them injected with webpack in parent project.

Reply:
So, same for var?